File: Models/JournalEntry.swift
import SwiftData

@Model
final class JournalEntry {
    var id: UUID
    var date: Date
    var text: String

    @Relationship(deleteRule: .cascade, inverse: \Plant.journalEntries)
    var plant: Plant?

    init(date: Date = Date(), text: String) {
        self.id = UUID()
        self.date = date
        self.text = text
    }
}

File: app/src/main/java/com/soilandsupper/domain/model/JournalEntry.kt
package com.soilandsupper.domain.model

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "journal_entries",
    indices = [Index(value = ["plantId"])],
    foreignKeys = [
        ForeignKey(
            entity = Plant::class,
            parentColumns = ["id"],
            childColumns = ["plantId"],
            onDelete = ForeignKey.SET_NULL
        )
    ]
)
data class JournalEntry(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val plantId: Long? = null,
    val date: Long = System.currentTimeMillis(),
    val text: String
)
